src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1265,81 +1268,176 @@ public void ForceRun()
-                        // The following logic to start periodic evaluation will need to be moved
-                        // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
-                        // as it causes an immediate crash due to the above ResetEngine call.
-                        var hws = ws as HomeWorkspaceModel;
-                        if (hws != null)
-                        {
-                            // TODO: #4258
-                            // Remove this ResetEngine call when multiple home workspaces is supported.
-                            // This call formerly lived in DynamoViewModel
-                            ResetEngine();
+          AddWorkspace(ws);
+          CurrentWorkspace = ws;
+        }
 
-                            if (hws.RunSettings.RunType == RunType.Periodic)
-                            {
-                                hws.StartPeriodicEvaluation();
-                            }
-                        }
+        private void SetPeriodicEvaluation(WorkspaceModel ws)
+        {
+          // TODO: #4258
+          // The following logic to start periodic evaluation will need to be moved
+          // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
+          // as it causes an immediate crash due to the above ResetEngine call.
+          var hws = ws as HomeWorkspaceModel;
+          if (hws != null)
+          {
+            // TODO: #4258
+            // Remove this ResetEngine call when multiple home workspaces is supported.
+            // This call formerly lived in DynamoViewModel
+            ResetEngine();
 
-                        CurrentWorkspace = ws;
-                        return;
-                    }
-                }
+            if (hws.RunSettings.RunType == RunType.Periodic)
+            {
+              hws.StartPeriodicEvaluation();
             }
-            Logger.LogError("Could not open workspace at: " + xmlPath);
+          }
+        }
+
+        private bool OpenJsonFile(WorkspaceInfo workspaceInfo, string fileContents, out WorkspaceModel workspace)
+        {
+            CustomNodeManager.AddUninitializedCustomNodesInPath(
+                Path.GetDirectoryName(workspaceInfo.FileName),
+                IsTestMode);
+
+            workspace = WorkspaceModel.FromJson(
+                fileContents, 
+                LibraryServices,
+                EngineController, 
+                Scheduler, 
+                NodeFactory, 
+                IsTestMode, 
+                false,
+                CustomNodeManager);
+            
+            workspace.FileName = workspaceInfo.FileName;
+
+            workspace.OnCurrentOffsetChanged(
+                this,
+                new PointEventArgs(new Point2D(workspaceInfo.X, workspaceInfo.Y)));
+
+            workspace.ScaleFactor = workspaceInfo.ScaleFactor;
+
+            // TODO: DeserializeObject does not appear to check a schema and so will not fail?
```

Comment:
@ColinDayOrg I think you'd want to use json.net jsonSchema to generate a schema from the class and then validate the json object against it, but I think that library requires a commercial license.